### ISSUE TITLE
Stop drag when mouse leaves window

### ISSUE THREAD

**Issue by cybergeek94** [PistonDevelopers/drag_controller#20]
If the mouse cursor leaves the window while a drag is in progress, the drag controller will continue the drag when it re-enters, even if the mouse button was released while the cursor was out of view. This is generally undesirable behavior, as far as I can tell.

**Comment by bvssvni** [comment]
Some window back-ends register mouse movements when the drag is outside the window. In this case you might want to continue the drag even the cursor leaves the window.

**Comment by cybergeek94** [comment]
I don't think we have support for mouse enter/leave events yet anyways. Just focus gain/loss, but that doesn't cover when the window is focused but the mouse is outside its bounds.

**Comment by bvssvni** [comment]
Perhaps we should add a second `event` method (call it something else) that takes event and a window as arguments and which interrupts dragging when leaving the window.

**Comment by bvssvni** [comment]
The current design allows you to continue dragging when interrupting by focus. To me it looks like interrupt by focus and interrupt by moving outside the window are two different things. You might choose to interrupt by one but not the other.

**Comment by bvssvni** [comment]
When thinking about window back-ends that does not have enter/leave events, I figured out that this should probably be solved in this library and not the window back-end https://github.com/PistonDevelopers/drag_controller/issues/21.

**Comment by bvssvni** [comment]
X = interrupt dragging
B = lost focus
C = leaved window
D = window back-end emits coordinates outside window

`B, ?X.` You usually want to interrupt dragging when loosing focus.

`D, C, ?!X` You usually don't want to interrupt dragging when leaving the window if the window back-end emits coordinates outside the window.

`!D, C, ?X` You usually want to interrupt dragging when leaving the window if the window back-end does not emit coordinates outside the window.

**Comment by bvssvni** [comment]
E = cancel dragging by clicking right mouse button
F = left mouse button was released outside the window

`E, !D, C, ?!X.` To me, it seems that you usually don't want to interrupt dragging if there is a way to cancel it when you enter the window again.

`E, !D, C, F, ?X.` It is only annoying when you release the button outside the window.
